csrc/fusion_segmenter.cpp

```diff
@@ -3567,7 +3581,10 @@ SegmentCandidateFinder::SegmentCandidateFinder(
     const KernelArgumentHolder& inputs,
     SegmentCandidateFinderOptions options)
     : options_(options),
-      runtime_info_(fusion.get(), inputs),
+      runtime_info_(
+          inputs.empty()
```

Comment:
Ok I see. Let me run the CI to see if it is triggered in the tests.

Anyways, let me know, but if it is not safe, then we can simply make `inputs` "optionnal". To do so I suggest we make it a raw pointer `KernelArgumentHolder*` and check if this ptr is null (because, correct me if I'm wrong, we cannot make `std::optionnal` of a reference, so raw ptr seems relevant here, but let me know if you suggest otherwise)

Reply:
If I am correct I see no relevant error in the CI builds. Let me know whether I should make a change here

Reply:
I'd include https://github.com/NVIDIA/Fuser/commit/5554c7c86e820b89d7f474ba7e3ea71273948294 to be safe. I guess we might not (but should) have a unit test for empty inputs. I'll add one. 

A non-blocking question: why doesn't the multidevice runtime feed in the args? Is there a problem to access runtime inputs in the multi-device setting?

Reply:
>  I guess we might not (but should) have a unit test for empty inputs. I'll add one.

+1

Reply:
Ok I will add this commit

> A non-blocking question: why doesn't the multidevice runtime feed in the args? Is there a problem to access runtime inputs in the multi-device setting?

Yes in some sense.
1) First of all, this segmentation step doesn't (at least for now) rely on the actual inputs. It can be done at the purely symbolic Fusion definition (+ parallel type information), so there is no real meaning to feed the inputs
2) In the multidevice setting in general, not all the devices necessarily have the full input. Think of a simple pipeline parallelism Fusion = stage0 -> stage1, then the devices in charge of stage1 won't have the Fusion's input

Reply:
I assume this will change in the future. Currently the multidevice scheduling is done manually, so the segmentation itself doesn't need any info about inputs, but in the future scheduling and segmentation would be both runtime decisions using input information, e.g., distribution strategy may depend on the size of the input. For now, and I suppose that's what Sam indicated too, this should be fine.